lib/pengine/unpack.c

```diff
@@ -4971,9 +4988,23 @@ add_node_attrs(const xmlNode *xml_obj, pcmk_node_t *node, bool overwrite,
                          cluster_name);
     }
 
-    pe__unpack_dataset_nvpairs(xml_obj, PCMK_XE_INSTANCE_ATTRIBUTES, &rule_data,
-                               node->priv->attrs, NULL, overwrite,
-                               scheduler);
+    if (overwrite) {
+        /* @TODO Try to reorder some unpacking so that we don't need the
+         * overwrite argument or to unpack into a temporary table
+         */
+        GHashTable *unpacked = pcmk__strkey_table(free, free);
+
+        pe__unpack_dataset_nvpairs(xml_obj, PCMK_XE_INSTANCE_ATTRIBUTES,
+                                   &rule_data, unpacked, NULL, FALSE,
+                                   scheduler);
```

Comment:
This will also change how same-scored blocks behave, but that's an improvement for consistency (and probably not possible for transient attributes without manual modification)

Reply:
Yeah I wrote this one off since the status section is considered private